dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFileTinyLocalIndexWriter.cpp

```diff
@@ -192,40 +175,19 @@ ColumnFileTinyPtr ColumnFileTinyLocalIndexWriter::buildIndexForFile(
         const auto & cd = read_columns->at(col_idx);
         for (const auto & index : index_builders[cd.id])
         {
-            switch (index.info.kind)
-            {
-            case TiDB::ColumnarIndexKind::Vector:
-            {
-                RUNTIME_CHECK(index.builder_vector);
-                auto index_page_id = options.storage_pool->newLogPageId();
-                MemoryWriteBuffer write_buf;
-                CompressedWriteBuffer compressed(write_buf);
-                index.builder_vector->finalize(compressed);
-                compressed.next();
-                auto data_size = write_buf.count();
-                auto buf = write_buf.tryGetReadBuffer();
-                // ColumnFileDataProviderRNLocalPageCache currently does not support read data with fields
-                options.wbs.log.putPage(index_page_id, 0, buf, data_size, {data_size});
-
-                auto idx_info = dtpb::ColumnFileIndexInfo{};
-                idx_info.set_index_page_id(index_page_id);
-                auto * idx_props = idx_info.mutable_index_props();
-                idx_props->set_kind(dtpb::IndexFileKind::VECTOR_INDEX);
-                idx_props->set_index_id(index.info.index_id);
-                idx_props->set_file_size(data_size);
-                auto * vector_index = idx_props->mutable_vector_index();
-                vector_index->set_format_version(0);
-                vector_index->set_dimensions(index.info.def_vector_index->dimension);
-                vector_index->set_distance_metric(
-                    tipb::VectorDistanceMetric_Name(index.info.def_vector_index->distance_metric));
-                index_infos->emplace_back(std::move(idx_info));
-
-                break;
-            }
-            default:
-                RUNTIME_CHECK_MSG(false, "Unsupported index kind: {}", magic_enum::enum_name(index.info.kind));
-                break;
-            }
+            RUNTIME_CHECK(index.index_writer);
+            auto index_page_id = options.storage_pool->newLogPageId();
+            MemoryWriteBuffer write_buf;
+            CompressedWriteBuffer compressed(write_buf);
+            dtpb::ColumnFileIndexInfo pb_cf_idx;
+            pb_cf_idx.set_index_page_id(index_page_id);
+            auto idx_info = index.index_writer->finalize(compressed, [&write_buf] { return write_buf.count(); });
+            pb_cf_idx.mutable_index_props()->Swap(&idx_info);
+            auto data_size = write_buf.count();
+            auto buf = write_buf.tryGetReadBuffer();
+            // ColumnFileDataProviderRNLocalPageCache currently does not support read data with fields
+            options.wbs.log.putPage(index_page_id, 0, buf, data_size, {data_size});
```

Comment:
What magnitude is the size of the page for the inverted index going to be?

Reply:
`AfterCompressed(MetaSize + BlockCount * BlockSize)`

Reply:
Why do we need to set `data_sizes` if we always read the whole page from disk?

Reply:
ColumnFileDataProviderRNLocalPageCache currently does not support read data withiout fields